crates/primitives/src/signature.rs

```diff
@@ -0,0 +1,855 @@
+
+impl From<bool> for Parity {
+    fn from(value: bool) -> Self {
+        Parity::Parity(value)
+    }
+}
+
+impl From<u64> for Parity {
+    fn from(value: u64) -> Self {
+        Parity::V(value)
+    }
+}
+
+impl Parity {
+    /// Get the chain_id of the V value, if any
+    pub const fn chain_id(&self) -> Option<ChainId> {
+        match *self {
+            Parity::V(mut v @ 35..) => {
+                if v % 2 == 0 {
+                    v -= 1;
+                }
+                v -= 35;
+                Some(v / 2)
+            }
+            _ => None,
+        }
+    }
+
+    /// Return the y-parity
+    pub fn y_parity(&self) -> bool {
+        match self {
+            Parity::V(v @ 0..=34) => v % 2 == 1,
+            Parity::V(v) => (v ^ 1) % 2 == 0,
+            Parity::Parity(y) => *y,
+        }
+    }
+
+    /// Return the y-parity as 0 or 1
+    pub fn y_parity_byte(&self) -> u8 {
+        self.y_parity() as u8
+    }
+
+    /// Invert the parity
+    pub fn inverted(&self) -> Self {
+        match self {
+            Parity::Parity(y) => Parity::Parity(!y),
+            Parity::V(v @ 0..=34) => Parity::V(if v % 2 == 0 { v - 1 } else { v + 1 }),
+            Parity::V(v @ 35..) => Parity::V(*v ^ 1),
+        }
+    }
+
+    /// Apply EIP 155 to the V value. This is a nop for parity values.
+    pub const fn with_chain_id(&self, chain_id: ChainId) -> Self {
+        let parity = match self {
+            Parity::V(v) => normalize_v_to_byte(*v) == 1,
+            Parity::Parity(y) => *y,
+        };
+
+        Self::V(to_eip155_v(parity as u8, chain_id))
+    }
+
+    #[cfg(feature = "k256")]
+    /// Determine the recovery id.
+    pub const fn recid(&self) -> k256::ecdsa::RecoveryId {
+        let recid_opt = match self {
+            Parity::V(v) => Some(normalize_v(*v)),
+            Parity::Parity(y) => k256::ecdsa::RecoveryId::from_byte(*y as u8),
+        };
+
+        // manual unwrap for const fn
+        match recid_opt {
+            Some(recid) => recid,
+            None => unreachable!(),
+        }
+    }
+}
+
+#[cfg(feature = "rlp")]
+impl alloy_rlp::Encodable for Parity {
+    fn encode(&self, out: &mut dyn alloy_rlp::BufMut) {
+        match self {
+            Parity::V(v) => v.encode(out),
+            Parity::Parity(false) => 27u8.encode(out),
+            Parity::Parity(true) => 28u8.encode(out),
+        }
+    }
+
+    fn length(&self) -> usize {
+        match self {
+            Parity::V(v) => v.length(),
+            Parity::Parity(_) => 1,
+        }
+    }
+}
+
+#[cfg(feature = "rlp")]
+impl alloy_rlp::Decodable for Parity {
+    fn decode(buf: &mut &[u8]) -> Result<Self, alloy_rlp::Error> {
+        let v = u64::decode(buf)?;
+        Ok(match v {
+            27 => Self::Parity(false),
+            28 => Self::Parity(true),
+            v => Self::from(v),
+        })
+    }
+}
+
+/// An Ethereum ECDSA signature.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct Signature<T: Copy + Eq> {
```

Comment:
Might need methods ported from here https://github.com/alloy-rs/alloy/pull/20/files#diff-097d91c911beebc907ad9aa358245953a05f69993d37ef026107af7c26d3c872

Reply:
most of those exist but are renamed iirc